examples/tv-app/linux/include/channel/ChannelManager.cpp

```diff
@@ -167,10 +167,10 @@ bool ChannelManager::HandleChangeChannelByNumber(const uint16_t & majorNumber, c
     return channelChanged;
 }
 
-bool ChannelManager::HandleSkipChannel(const uint16_t & count)
+bool ChannelManager::HandleSkipChannel(const int16_t & count)
 {
     // TODO: Insert code here
-    uint16_t newChannelIndex = static_cast<uint16_t>((count + mCurrentChannelIndex) % mChannels.size());
+    uint16_t newChannelIndex = static_cast<int16_t>((count + mCurrentChannelIndex) % mChannels.size());
```

Comment:
This is a bit more complicated.  First of all, you do want to end up casting to `uint16_t` at the end, because the result is supposed to be `uint16_t`.

Second, I am 99% sure this formula is broken if `count` is allowed to be negative and it it happens that `count` is negative and larger than `mCurrentChannelIndex` by absolute value. For example, let's say `mCurrentChannelIndex == 5`, `count == -7`, and `size == 7`.  The expectation would be that we end up with 5, right?  But the formula will produce, on a system where `size_t` is 64-bit:

1. `count  + mCurrentChannelIndex == -2`, of type `int` due to integer promotion.
2. `((int)-2) / mChannels.size() == 0x2492492492492492` because the `-2` gets cast to `uint64_t` and then unsigned division happens.
3. `0x2492492492492492 * ((int)-2) == 0xfffffffffffffffe` (which happens to be `-2` cast to `uint64_t`)
4. `count  % mCurrentChanntlIndex == 0` because that's how `%` is defined in C++: our division had no remainder.

That's because `2^3 == 1 mod 7`, so `2^64 == 2 mod 7`, and hence `-2` is divisible by 7 if the 7 is a 64-bit `size_t`.  There are similar weird effects if `size_t` is 32-bit: in that case `2^32 == 4 mod 7`, so -2 ends up being promoted to a value that is `2 mod 7` and then we end up with `2` instead of `5` as our final result.

Please:

1) Rewrite this to produce correct results.
2) Add unit tests that test this for a variety of inputs and `mChannels` sizes.